sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/HyperLogLogPlusPlus.scala

```diff
@@ -447,6 +447,7 @@ object HyperLogLogPlusPlus {
 
   private def validateDoubleLiteral(exp: Expression): Double = exp match {
     case Literal(d: Double, DoubleType) => d
+    case Literal(dec: Decimal, dt: DecimalType) => dec.toDouble
```

Comment:
I changed this because the old SQL parser was passing Decimal to HyperLogLogPlusPlus.

The pattern match eliminates the possibility of a NPE. Spark ```Decimal``` delegates the ```toDouble``` call to java ```BigDecimal``` which will return ```(-)Infinity``` if the decimal is outside of the double range. We should be good here.